modules/postgresql/main.tf

```diff
@@ -0,0 +1,106 @@
+terraform {
+  required_version = ">=1.1.2"
+
+  required_providers {
+    azurerm = "=2.94.0"
+  }
+
+  backend "azurerm" {}
+}
+
+provider "azurerm" {
+  features {}
+}
+
+resource "random_password" "postgresql_admin" {
+  length           = 16
+  special          = false
+  override_special = "_%@"
+  keepers          = var.password_keeper
+}
+
+resource "azurerm_postgresql_flexible_server" "postgresql_server" {
+  name                = var.name
+  location            = var.location
+  resource_group_name = var.resource_group_name
+
+  sku_name = var.postgresql_sku_name
+
+  storage_mb            = var.postgresql_db_size
+  backup_retention_days = 30
+
+  administrator_login    = var.postgresql_username
+  administrator_password = random_password.postgresql_admin.result
+  version                = var.postgresql_version
+  zone                   = "1"
+  delegated_subnet_id    = var.subnet_id
```

Comment:
Please use delegated_subnet_id, since more subnet_id's may be required for private endpoints